pallets/attestation/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2023 Entropy Cryptography Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! # Attestation Pallet
+
+#![cfg_attr(not(feature = "std"), no_std)]
+pub use pallet::*;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+
+pub mod weights;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use entropy_shared::QuoteInputData;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+    use sp_std::vec::Vec;
+    use tdx_quote::Quote;
+
+    pub use crate::weights::WeightInfo;
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_staking_extension::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// Describes the weights of the dispatchables exposed by this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::genesis_config]
+    #[derive(frame_support::DefaultNoBound)]
+    pub struct GenesisConfig<T: Config> {
+        pub initial_pending_attestations: Vec<(T::AccountId, [u8; 32])>,
+        pub initial_attestation_requests: Vec<(BlockNumberFor<T>, Vec<Vec<u8>>)>,
```

Comment:
```suggestion
        pub initial_attestation_requests: Vec<(BlockNumberFor<T>, Vec<T::AccountId>)>,
```

Why not do something like this instead?

Reply:
Good point. I thought it is not possible to have a <T::AccountId> as a value in a `StorageMap`.  As in all our other pallets, we use `Vec<u8>`.  But actually i just tried it, and it compiled fine.

So now the issue is that in `entropy_shared::OcwMessageAttestationRequest` we represent this as a `[u8; 32]`. Since theres no way to turn a generic `T::AccountId` into a `[u8; 32]`, we would have to make `OcwMessageAttestationRequest` generic for chain config.

I'd be happy to do this, but i have the feeling it there might be issues with deriving `Serialize` and `Deserialize` and probably it would make sense to also do it to all the other types we have in `entropy-shared` which represent account ids with `Vec<u8>`. So maybe its for another PR.